### PR TITLE
Add operator IDs into the text based circuit draw function.

### LINKED ISSUE #1397: Display `id` of gates when printing with `qml.draw`
With the new functionality to assign "names" to gates, it might be worthwhile to have this information propagate to the printed circuit of `qml.draw`. I'm thinking something like
```
def circuit(x,d):
    qml.RX(d, id="data", wires=0)
    qml.Rot(x, id="parameters", wires=0)
   return qml.expval(qml.PauliZ(0)) 
   
d = 0.123
x = np.array([0.863, 0.611, 0.281])
print(qml.draw(circuit)(x,d))
0: ──RX(0.123, "data")──Rot(0.863, 0.611, 0.281, "parameters")──┤ ⟨Z⟩
```

We could add an keyword argument to `qml.draw` to expose or suppress this info during printing

### PR DESCRIPTION
**Pull request by JustinWoodring**
**Context:**
Per the associated issue "With the new functionality to assign "names" to gates, it might be worthwhile to have this information propagate to the printed circuit of qml.draw."

**Description of the Change:**
Modifies the printed text output resulting from qml.draw() to indicate the ids of operators.

```python
def circuit(x,d):
    qml.RX(d, id="data", wires=0)
    qml.Rot(x, wires=0)
    return qml.expval(qml.PauliZ(0)) 
   
d = 0.123
x = np.array([0.863, 0.611, 0.281])
print(qml.draw(circuit)(*x,d))
```
Output:
```0: ──RX(0.123,"data")──Rot(0.863,0.611,0.281)──┤ ⟨Z⟩```

**Benefits:**
This allows users to see operators displayed with their ids in the printed circuit output.

**Possible Drawbacks:**
Too many rendered operators with ID's might consume more space causing the printed circuit to wrap more.

**Related GitHub Issues:**
This implements and closes #1397

### CODE REVIEW

**Review by albi3ro — APPROVED** [review]
Looks great to me 🎉  Nice to close such a backlogged issue.

No comments or concerns from my side. I'm happy to approve :)

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
 * `qml.ArbitraryUnitary` now supports batching.
   [(#4745)](https://github.com/PennyLaneAI/pennylane/pull/4745)
 
+* `qml.draw` now renders operator ids.
```
[the remark this answers is not in the document]
Comment (JustinWoodring) [reply doc/releases/changelog-dev.md:15]: Perfect! Thanks for the suggestion and review :)

**Comment by trbromley** [comment]
Thanks @JustinWoodring for your contribution! :trophy:

This should be released as part of v0.34 in early January. If you would like us to tag you as part of our Twitter/X marketing, feel free to optionally share your handle here.

**Comment by JustinWoodring** [comment]
I don't use Twitter, but if you do similar stuff on LinkedIn, my LinkedIn is, https://www.linkedin.com/in/justin-woodring/ 😃

**Comment by trbromley** [comment]
Thanks!

**Pull request merged**